pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -570,29 +573,78 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileLogStorage) elasticsearchSecrets(ctx context.Context) ([]*corev1.Secret, error) {
-	var secrets []*corev1.Secret
+func (r *ReconcileLogStorage) getCertificateKeySecret(ctx context.Context, keySecretName string, keySecretNs string, pubSecretName string, pubSecretNs string, svcDNSNames ...string) (*corev1.Secret, error) {
 	secret := &corev1.Secret{}
-	if err := r.client.Get(ctx, types.NamespacedName{Name: render.TigeraElasticsearchCertSecret, Namespace: render.OperatorNamespace()}, secret); err != nil {
-		if errors.IsNotFound(err) {
-			svcDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, r.clusterDomain)
-			secret, err = render.CreateOperatorTLSSecret(nil,
-				render.TigeraElasticsearchCertSecret, "tls.key", "tls.crt",
-				render.DefaultCertificateDuration, nil, svcDNSNames...,
-			)
-		} else {
+
+	createCert := func() (*corev1.Secret, error) {
```

Comment:
I don't think we should use an anonymous function here, if it's useful to do this here it'll be useful else where and should be a function that can be accessed by all.